components/job-orchestration/job_orchestration/scheduler/scheduler_data.py

```diff
@@ -35,12 +40,18 @@ class InternalJobState(Enum):
     RUNNING = auto()
 
 
-class QueryJob(BaseModel):
+class QueryJob(BaseModel, ABC):
     id: str
     state: InternalJobState
     start_time: Optional[datetime.datetime]
     current_sub_job_async_task_result: Optional[Any]
 
+    @abstractmethod
+    def type(self) -> QueryJobType: ...
+
+    @abstractmethod
+    def job_config(self) -> QueryConfig: ...
+
 
 class SearchJob(QueryJob):
     search_config: SearchConfig
```

Comment:
Should we rename this to just "job_config"

Reply:
I think search_config is fine.